platform/binance/model.go

```diff
@@ -31,101 +32,104 @@ type Error struct {
-	SubTxsDto     SubTxsDto   `json:"subTxsDto"`
+	Asset       string `json:"txAsset"`
+	BlockHeight uint64 `json:"blockHeight"`
+	Code        int    `json:"code"`
+	Data        string `json:"data"`
+	Fee         string `json:"txFee"`
+	FromAddr    string `json:"fromAddr"`
+	Memo        string `json:"memo"`
+	OrderID     string `json:"orderId"`
+	Sequence    uint64 `json:"sequence"`
+	Source      int    `json:"source"`
+	Timestamp   string `json:"timeStamp"`
+	ToAddr      string `json:"toAddr"`
+	TxHash      string `json:"txHash"`
+	Type        TxType `json:"txType"`
+	Value       string `json:"value"`
+}
+
+type BlockTransactions struct {
+	BlockHeight int64  `json:"blockHeight"`
+	Txs         []TxV2 `json:"tx"`
 }
 
-type SubTxsDto struct {
-	TotalNum     uint   `json:"totalNum"`
-	SubTxDtoList SubTxs `json:"subTxDtoList"`
+type TxV2 struct {
+	Tx
+	OrderID         string  `json:"orderId"`         // Optional. Available when the transaction type is NEW_ORDER
+	SubTransactions []SubTx `json:"subTransactions"` // Optional. Available when the transaction has sub-transactions, such as multi-send transaction or a transaction have multiple assets
 }
 
 type SubTx struct {
-	Hash     string      `json:"hash"`
-	Height   uint64      `json:"height"`
-	Type     TxType      `json:"type"`
-	Value    json.Number `json:"value"`
-	Asset    string      `json:"asset"`
-	FromAddr string      `json:"fromAddr"`
-	ToAddr   string      `json:"toAddr"`
-	Fee      json.Number `json:"fee"`
+	Asset    string `json:"txAsset"`
+	Height   uint64 `json:"blockHeight"`
+	Fee      string `json:"txFee"`
+	FromAddr string `json:"fromAddr"`
+	Hash     string `json:"txHash"`
+	ToAddr   string `json:"toAddr"`
+	Type     TxType `json:"txType"`
+	Value    string `json:"value"`
 }
 
-type SubTxs []SubTx
+type TokenList []Token
 
-func (subTxs *SubTxs) getTxs() (txs []Tx) {
-	mapTx := map[string]Tx{}
-	for _, subTx := range *subTxs {
-		key := subTx.ToAddr + subTx.Asset
-		tx, ok := mapTx[key]
-		if !ok {
-			mapTx[key] = subTx.toTx()
-			continue
-		}
-		txValue, err := tx.Value.Float64()
-		if err != nil {
-			txValue = 0
-		}
-		subTxValue, err := subTx.Value.Float64()
-		if err != nil {
-			subTxValue = 0
-		}
-		value := numbers.Float64toString(txValue + subTxValue)
-		tx.Value = json.Number(value)
-		mapTx[key] = tx
+type Token struct {
+	Name           string `json:"name"`
+	OriginalSymbol string `json:"original_symbol"`
+	Owner          string `json:"owner"`
+	Symbol         string `json:"symbol"`
+	TotalSupply    string `json:"total_supply"`
+}
+
+func (tx *Tx) getFee() string {
+	fee := "0"
+	if _, err := strconv.ParseFloat(tx.Fee, 64); err == nil {
+		fee = numbers.DecimalExp(tx.Fee, 8)
 	}
-	for _, tx := range mapTx {
-		txs = append(txs, tx)
+	return fee
+}
+
+func (tx *Tx) getStatus() blockatlas.Status {
+	switch tx.Code {
+	case 0:
+		return blockatlas.StatusCompleted
+	default:
+		return blockatlas.StatusError
+	}
+}
+
+func (tx *Tx) getError() string {
```

Comment:
Maybe return bool value?
Like that:

```
func (tx *Tx) isCompleted() bool {
switch tx.getStatus() {
	case blockatlas.StatusCompleted:
		return true
	default:
		return false
    }
}
```

Reply:
Error is type `string`, i is it make sense to return string as is it now.
<img width="847" alt="Screen Shot 2020-05-03 at 10 18 32 PM" src="https://user-images.githubusercontent.com/3277207/80937954-08c09c80-8d8c-11ea-9e97-f009b77cf3b3.png">